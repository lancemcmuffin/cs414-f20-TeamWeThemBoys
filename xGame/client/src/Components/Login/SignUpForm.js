import React from "react";
import PropTypes from "prop-types";
import { Row, Col } from "reactstrap";
import { Formik } from "formik";
import { Button, TextField } from "@material-ui/core";

const SignUpForm = (props) => {
  const { setShowLoginForm, onSignUp } = props;

  return (
    <Formik
      onSubmit={(values) => {
        console.log("TODO: Hook up registration API", values);
        onSignUp();
      }}
      initialValues={{
        email: "",
        username: "",
        password: ""
      }}
    >
      {(formProps) => {
        const { setFieldValue, values } = formProps;
        return (
          <Col>
            <Row>
              <Col>
                <TextField
                  color="secondary"
                  id="username"
                  variant="outlined"
                  style={{ width: "80%" }}
                  name="username"
                  label="Username"
                  type="username"
                  value={values?.["username"] || ""}
                  onChange={(e) => setFieldValue("username", e.target.value)}
                />
              </Col>
            </Row>
            <Row className="pt-2">
              <Col>
                <TextField
                  color="secondary"
                  id="email"
                  variant="outlined"
                  style={{ width: "80%" }}
                  name="email"
                  label="Email Address"
                  value={values?.["email"] || ""}
                  onChange={(e) => setFieldValue("email", e.target.value)}
                />
              </Col>
            </Row>
            <Row className="pt-2">
              <Col>
                <TextField
                  color="secondary"
                  id="password"
                  variant="outlined"
                  style={{ width: "80%" }}
                  name="password"
                  label="Password"
                  type="password"
                  value={values?.["password"] || ""}
                  onChange={(e) => setFieldValue("password", e.target.value)}
                />
              </Col>
            </Row>
            <Row className="pt-2">
              <Col className="d-flex justify-content-center">
                <Button
                  color="secondary"
                  variant="contained"
                  style={{ textTransform: "none" }}
                  onClick={formProps.handleSubmit}
                >
                  Sign Up
                </Button>
              </Col>
            </Row>
            <Row className="pt-2">
              <Col className="d-flex justify-content-center">
                <div>Already have an account? </div>
                <Button
                  size="small"
                  color="secondary"
                  style={{ textTransform: "none" }}
                  onClick={() => setShowLoginForm(true)}
                >
                  Log In
                </Button>
              </Col>
            </Row>
          </Col>
        );
      }}
    </Formik>
  );
};

SignUpForm.propTypes = {
  setShowLoginForm: PropTypes.func,
  onSignUp: PropTypes.func
};

SignUpForm.defaultProps = {
  setShowLoginForm: () => {},
  onSignUp: () => {}
};

export default SignUpForm;
