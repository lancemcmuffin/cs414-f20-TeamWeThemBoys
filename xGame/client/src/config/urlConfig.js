/**
 * @module urlConfig.js
 * @description exports URL constants used by services
 */

const localhostURL = "http://localhost:8080";
const serverURL = "tbd";

const serviceURL = localhostURL;

export const matchServiceURL = `${serviceURL}/match`;
export const userServiceURL = `${serviceURL}/user`;
export const messageServiceURL = `${serviceURL}/message`;
export const gameServiceURL = `${serviceURL}/game`;
