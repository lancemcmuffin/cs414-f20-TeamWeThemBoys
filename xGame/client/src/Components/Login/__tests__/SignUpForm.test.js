import "../../../enzyme.config.js";
import React from "react";
import { shallow } from "enzyme";

import SignUpForm from "../SignUpForm";

function testInitialState() {
  const app = shallow(<SignUpForm />);
  expect(app.html()).toMatchSnapshot();
}

test("Testing login initial state", testInitialState);
