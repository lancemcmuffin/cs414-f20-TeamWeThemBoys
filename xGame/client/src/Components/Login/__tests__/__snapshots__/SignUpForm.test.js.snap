// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login initial state 1`] = `"<div class=\\"col\\"><div class=\\"row\\"><div class=\\"col\\"><div class=\\"MuiFormControl-root MuiTextField-root\\" style=\\"width:80%\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"username\\" id=\\"username-label\\">Username</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input type=\\"username\\" aria-invalid=\\"false\\" id=\\"username\\" name=\\"username\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Username</span></legend></fieldset></div></div></div></div><div class=\\"pt-2 row\\"><div class=\\"col\\"><div class=\\"MuiFormControl-root MuiTextField-root\\" style=\\"width:80%\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"email\\" id=\\"email-label\\">Email Address</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"email\\" name=\\"email\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Email Address</span></legend></fieldset></div></div></div></div><div class=\\"pt-2 row\\"><div class=\\"col\\"><div class=\\"MuiFormControl-root MuiTextField-root\\" style=\\"width:80%\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"password\\" id=\\"password-label\\">Password</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input type=\\"password\\" aria-invalid=\\"false\\" id=\\"password\\" name=\\"password\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Password</span></legend></fieldset></div></div></div></div><div class=\\"pt-2 row\\"><div class=\\"d-flex justify-content-center col\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary\\" tabindex=\\"0\\" type=\\"button\\" style=\\"text-transform:none\\"><span class=\\"MuiButton-label\\">Sign Up</span></button></div></div><div class=\\"pt-2 row\\"><div class=\\"d-flex justify-content-center col\\"><div>Already have an account? </div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-textSizeSmall MuiButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" style=\\"text-transform:none\\"><span class=\\"MuiButton-label\\">Log In</span></button></div></div></div>"`;
